File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.java
// Last updated: 2/22/2026, 1:55:08 PM
class Solution {
    public int removeDuplicates(int[] nums) {
        int l = 1, r = 1;
        int n = nums.length;
        while(r<n){
            if(nums[r]!=nums[r-1]){
                nums[l] = nums[r];
                l++;
            }
            r++;
        }
        return l; 
    }
}

File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.cpp
// Last updated: 3/30/2026, 8:44:34 AM
class Solution {
private:
    void swap(vector<int>& nums,int slow, int fast){
        int temp = nums[slow];
        nums[slow] = nums[fast];
        nums[fast] = temp;
    }
public:
    int removeDuplicates(vector<int>& nums) {
        int slow = 0, fast = 1,n = nums.size();
        while(fast < n){
            if(nums[slow]!=nums[fast]){
                slow++;
                swap(nums,slow,fast);
            }
            fast++;
        }
        return slow+1;
    }
};